insights/parsers/snmp.py

```diff
@@ -0,0 +1,112 @@
+    ... '''.strip()
+    >>> from insights.tests import context_wrap
+    >>> shared = {TcpIpStats: TcpIpStats(context_wrap(SNMP_CONTENT))}
+    >>> stats = shared[TcpIpStats]
+    >>> snmp_stats = stats.get_nstats("Ip")
+    >>> print snmp_stats["DefaultTTL"]
+    64
+    >>> snmp_stats = stats.get_nstats("Udp")
+    >>> print snmp_stats["InDatagrams"]
+    18905
+"""
+
+from .. import Parser, parser, LegacyItemAccess
+
+
+def get_snmp_stats(nkeys, line_split):
+    nstats = {}
+    nstats = dict(zip(nkeys, map(int, line_split)))
+    return nstats
+
+
+@parser('proc_snmp_ipv4')
+class TcpIpStats(Parser, LegacyItemAccess):
+    """
+    Parser for ``/proc/net/snmp`` file.
+
+    Sample contents of file::
+
+        Ip: Forwarding DefaultTTL InReceives InHdrErrors InAddrErrors ForwDatagrams InUnknownProtos InDiscards InDelivers OutRequests OutDiscards OutNoRoutes ReasmTimeout ReasmReqds ReasmOKs ReasmFails FragOKs FragFails FragCreates
+        Ip: 2 64 2628 0 2 0 0 0 2624 1618 0 0 0 0 0 10 0 0 0
+        Icmp: InMsgs InErrors InDestUnreachs InTimeExcds InParmProbs InSrcQuenchs InRedirects InEchos InEchoReps InTimestamps InTimestampReps InAddrMasks InAddrMaskReps OutMsgs OutErrors OutDestUnreachs OutTimeExcds OutParmProbs OutSrcQuenchs OutRedirects OutEchos OutEchoReps OutTimestamps OutTimestampReps OutAddrMasks OutAddrMaskReps
+        Icmp: 0 0 0 0 0 0 0 0 0 0 0 0 0 2 0 2 0 0 0 0 0 0 0 0 0 0
+        IcmpMsg: InType3 OutType3
+        IcmpMsg: 34 44
+        Tcp: RtoAlgorithm RtoMin RtoMax MaxConn ActiveOpens PassiveOpens AttemptFails EstabResets CurrEstab InSegs OutSegs RetransSegs InErrs OutRsts
+        Tcp: 1 200 120000 -1 25 4 0 0 1 2529 1520 1 0 9
+        Udp: InDatagrams NoPorts InErrors OutDatagrams RcvbufErrors SndbufErrors
+        Udp: 95 0 0 95 1 4
+        UdpLite: InDatagrams NoPorts InErrors OutDatagrams RcvbufErrors SndbufErrors
+        UdpLite: 0 10 0 0 0 100
+
+    Resultant Data::
+
+        {
+            'Ip':
+                {
+                    'FragCreates': 0,
+                    'ReasmFails': 10,
+                    'Forwarding': 2,
+                    'ReasmOKs': 0,
+                    'ReasmReqds': 0,
+                    'ReasmTimeout': 0,
+                    ...
+                    ...
+                },
+            'Icmp':
+                {
+                    'InRedirects': 0,
+                    'InMsgs': 0,
+                    'InSrcQuenchs': 0,
+                    ...
+                    ...
+                }
+            ...
+            ...
+        }
+    """
+    def parse_content(self, content):
+        snmp_stats = {}
+        data_id = []
+        for line in content:
+            line_split = line.split()
+            if line_split[1].isdigit():
+                snmp_stats[line_split[0].replace(":", "")] = get_snmp_stats(data_id, line_split[1:])
+            else:
+                data_id = line_split[1:]
+        self.data = snmp_stats
+
+    def get_nstats(self, layer):
```

Comment:
Since the default `get()` method of the `LegacyItemsAccess` already implemented the logic of this `get_nstats()`. Do you think it's worth being added here?

~~~
"stats.get('Ip')" is totally equal to "stats.get_nstats('Ip')"
~~~